scalableminds/webknossos
# Auto switch organization suggestion

This PR makes it easier to view datasets of other organizations than the user's active organization. It adds a button to the "Failed loading dataset/tracing view" that lets the user directly switch to the correct organization that owns the dataset. But this only works if the user is actually part of the organization.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test: Better test this locally
- Turn on `isDemoInstance` in the application.conf (and restart your local running wk instance)
- Create a new user. This user should automatically have its own organization. Upload a test dataset for this organization.
- View the newly created dataset and copy the view link.
- Switch back to scmboy and try to view the dataset via the previously copied link.
- The loading should fail and there should be a second button (left to the usual "return to dashboard button") that should automatically switch to the correct organization, so that the dataset can be viewed.
- Then open an explorative tracing with scmboy on the dataset of the new organization.
- In the tracing view switch to the sample_organization.
- The tracing should fail loading and there should be the same suggestion button to switch to the correct organization as in the step before.
- Create a task for the dataset of the new organization and open it with the scmboy. When the scmboy has the new organization active, everything should be working fine. If the sample_organization is active, there should be the suggestion to switch the new organization.
- Now log in as the second user that was previously created and try to view a dataset, an explorational tracing and a task from a dataset from the sample_organization. In all three cases there shouldn't be a suggestion to switch the active ogranization.

### Issues:
- fixes #5511
- Note: This fix only works for view mode, as I do not know how to get the owning organization of the dataset for an e.g. explorative tracing.

------
- ~~[ ] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased) <-- This is an improvement usually just for us, right?~~
